File: app/src/main/java/com/example/hoavo/karaoke/views/CustomTextviewChannel.java
package com.example.hoavo.karaoke.views;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.support.annotation.Nullable;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.AppCompatTextView;
import android.util.AttributeSet;

import com.example.hoavo.karaoke.R;

/**
 * Copyright © 2017 AsianTech inc.
 * Created by rimoka on 31/07/2017.
 */

public class CustomTextviewChannel extends AppCompatTextView {

    private Paint mPaint;

    public CustomTextviewChannel(Context context) {
        super(context);
    }

    public CustomTextviewChannel(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public CustomTextviewChannel(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        if (mPaint == null) {
            mPaint = new Paint();
            mPaint.setColor(ContextCompat.getColor(getContext(), R.color.colorGrayLight));
            mPaint.setStrokeWidth(3);
        }

        canvas.drawLine(0, 0, getWidth(), 0, mPaint);
        canvas.drawLine(0, 0, 0, getHeight(), mPaint);
        canvas.drawLine(0, getHeight(), getWidth(), getHeight(), mPaint);
        canvas.drawLine(getWidth(), 0, getWidth(), getHeight(), mPaint);
    }
}

File: app/src/main/java/com/example/hoavo/kotlin/views/CustomTextviewChannel.kt
package com.example.hoavo.kotlin.views

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.support.v4.content.ContextCompat
import android.support.v7.widget.AppCompatTextView
import android.util.AttributeSet

import com.example.hoavo.karaoke.R

/**
 * Copyright © 2017 AsianTech inc.
 * Created by rimoka on 31/07/2017.
 */

class CustomTextviewChannel : AppCompatTextView {

    private val mPaint: Paint by lazy {
        Paint()
    }

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        mPaint.color = ContextCompat.getColor(context, R.color.colorGrayLight)
        mPaint.strokeWidth = 3f

        canvas.drawLine(0f, 0f, width.toFloat(), 0f, mPaint)
        canvas.drawLine(0f, 0f, 0f, height.toFloat(), mPaint)
        canvas.drawLine(0f, height.toFloat(), width.toFloat(), height.toFloat(), mPaint)
        canvas.drawLine(width.toFloat(), 0f, width.toFloat(), height.toFloat(), mPaint)
    }
}
